firmware/common2015/drivers/ring-buffer/ring_buffer.h

```diff
@@ -0,0 +1,201 @@
+/*

```

Comment:
Any reason for using this over [boost::circular_buffer](http://www.boost.org/doc/libs/1_58_0/doc/html/circular_buffer.html)?  The boost one is nice because it's C++ and we don't have to copy the source files into our repo.